script/Deploy.s.sol

```diff
@@ -2,54 +2,50 @@
 pragma solidity ^0.8.0;
 
 import {BaseScript} from "./Base.sol";
-
 import {PluginSettings} from "../src/utils/PluginSettings.sol";
 import {StagedProposalProcessorSetup as SPPSetup} from "../src/StagedProposalProcessorSetup.sol";
 
 import {PluginRepo} from "@aragon/osx/framework/plugin/repo/PluginRepo.sol";
 import {PermissionLib} from "@aragon/osx/core/permission/PermissionLib.sol";
 import {PluginRepoFactory} from "@aragon/osx/framework/plugin/repo/PluginRepoFactory.sol";
 
-import "forge-std/console.sol";
+import {console} from "forge-std/console.sol";
 
 contract Deploy is BaseScript {
     function run() external {
         // get deployed contracts
-        (pluginRepoFactory, managementDao) = getRepoContractAddresses(network);
+        pluginRepoFactory = getRepoFactoryAddress();
+        managementDao = getManagementDaoAddress();
 
         vm.startBroadcast(deployerPrivateKey);
 
         // crete plugin repo and version
-        (sppSetup, sppRepo) = _createPluginRepoAndVersion();
+        sppRepo = _createPluginRepo();
+        sppSetup = _createAndCheckNewVersion();
 
         //transfer ownership of the plugin to the management DAO and revoke from deployer
         _transferOwnershipToManagementDao();
 
         vm.stopBroadcast();
     }
 
-    function _createPluginRepoAndVersion()
-        internal
-        returns (SPPSetup _sppSetup, PluginRepo _sppRepo)
-    {
+    function _createPluginRepo() internal returns (PluginRepo _sppRepo) {
         // create plugin repo
         _sppRepo = PluginRepoFactory(pluginRepoFactory).createPluginRepo(
             PluginSettings.PLUGIN_REPO_ENS_SUBDOMAIN_NAME,
             deployer
         );
 
-        _sppSetup = new SPPSetup();
+        if (_sppRepo == PluginRepo(address(0))) {
```

Comment:
Would this ever happen ?

Reply:
probably never, should I remove the check?

Reply:
yes..